src/shell.js

```diff
@@ -276,7 +276,7 @@ if (ENVIRONMENT_IS_NODE) {
 if (ENVIRONMENT_IS_SHELL) {
 
 #if ENVIRONMENT && ASSERTIONS
```

Comment:
what's this change for?

Reply:
This fixes the detection of node in ES6 mode.  I can split this out potentially along with the test that needs it.    Might also be worth refactoring this into a macro since its used in a few different places.

Reply:
Split out into #24354